app/src/main/java/com/aptoide/uploader/account/view/AutoLoginPresenter.java

```
@@ -0,0 +1,123 @@
+  private Scheduler viewScheduler;
+
+  public AutoLoginPresenter(AutoLoginView view, AptoideAccountManager accountManager,
+      UploaderAnalytics uploaderAnalytics, AutoLoginManager autoLoginManager,
+      AutoLoginNavigator navigator, CompositeDisposable compositeDisposable,
+      Scheduler viewScheduler) {
+    this.view = view;
+    this.accountManager = accountManager;
+    this.uploaderAnalytics = uploaderAnalytics;
+    this.autoLoginManager = autoLoginManager;
+    this.navigator = navigator;
+    this.compositeDisposable = compositeDisposable;
+    this.viewScheduler = viewScheduler;
+  }
+
+  @Override public void present() {
+    showUserInfo();
+    checkLoginStatus();
+    handleAutoLogin();
+    handleOtherLoginsClick();
+    clearDisposable();
+  }
+
+  private void showUserInfo() {
+    view.showStrings();
+    view.showAvatar();
+  }
+
+  private void checkLoginStatus() {
+    compositeDisposable.add(view.getLifecycleEvent()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .flatMap(__ -> accountManager.getAccount())
+        .observeOn(viewScheduler)
+        .flatMap(account -> {
+          if (account.isLoggedIn()) {
+            if (account.hasStore()) {
+              navigator.navigateToMyAppsView();
+            } else {
+              navigator.navigateToCreateStoreView();
+            }
+          }
+          return Observable.empty();
+        })
+        .subscribe(__ -> {
+        }, throwable -> {
+          throwable.printStackTrace();
+        }));
+  }
+
+  private void handleAutoLogin() {
+    compositeDisposable.add(view.getLifecycleEvent()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .flatMap(created -> view.clickAutoLogin())
+        .observeOn(viewScheduler)
+        .flatMap(__ -> {
+          accountManager.logout();
+          accountManager.removeAccessTokenFromPersistence();
+          return tryAutoLogin();
+        })
+        .subscribe());
```

This will silently eat the error because onError is not implemented

---

I'll add a throwable